apps/src/templates/instructions/InstructionsCSF.jsx

```diff
@@ -323,61 +203,43 @@ class InstructionsCSF extends React.Component {
    * the height of the little icon if we're not collapsed
    */
   getMinHeight(collapsed = this.props.collapsed) {
-    const collapseButtonHeight = getOuterHeight(this.collapser, true);
-    const scrollButtonsHeight =
-      !collapsed && this.scrollButtons ? this.scrollButtons.getMinHeight() : 0;
-
-    const minIconHeight = this.icon ? getOuterHeight(this.icon, true) : 0;
-    const instructionsHeight = Math.min(
-      getOuterHeight(this.instructions, true),
-      this.props.maxHeight
-    );
-    const minInstructionsHeight =
-      this.props.collapsed ||
-      this.props.overlayVisible ||
-      this.props.isEmbedView
-        ? instructionsHeight
-        : 0;
+    const {leftColHeight, rightColHeight} = this.state;
+    const middleColHeight = this.getMinInstructionsHeight(collapsed);
 
-    const domNode = $(ReactDOM.findDOMNode(this));
-    const margins = domNode.outerHeight(true) - domNode.outerHeight(false);
-
-    const leftColHeight = minIconHeight;
-    const middleColHeight = minInstructionsHeight;
-    const rightColHeight = collapseButtonHeight + scrollButtonsHeight;
     return (
       Math.max(leftColHeight, middleColHeight, rightColHeight) +
-      margins +
-      HEADER_HEIGHT +
-      RESIZER_HEIGHT
+      this.contentWrapperHeight()
     );
   }
 
+  getMinInstructionsHeight = (collapsed = this.props.collapsed) => {
+    if (collapsed || this.props.overlayVisible || this.props.isEmbedView) {
+      return Math.min(this.getMiddleColHeight(), this.props.maxHeight);
+    } else {
+      return 0;
+    }
+  };
+
+  getMarginsHeight() {
+    const domNode = $(ReactDOM.findDOMNode(this));
```

Comment:
would it be hard to update this to use a ref instead of `findDOMNode`? you could attach it to the outer div returned by `render`...